aquadoggo/src/libp2p/config.rs

```diff
@@ -0,0 +1,163 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use std::path::PathBuf;
+
+use anyhow::Result;
+use libp2p::identity::Keypair;
+use libp2p::swarm::ConnectionLimits;
+use libp2p::Multiaddr;
+use log::info;
+use serde::{Deserialize, Serialize};
+
+use crate::libp2p::identity::Identity;
+
+/// Keypair file name.
+const KEYPAIR_FILE_NAME: &str = "libp2p.pem";
+
+/// QUIC default transport port.
+const QUIC_PORT: u16 = 48648;
+
+/// Libp2p config for the node.
+#[derive(Debug, Clone, Deserialize, Serialize)]
+pub struct Libp2pConfiguration {
+    /// Dial concurrency factor.
+    ///
+    /// Number of addresses concurrently dialed for a single outbound
+    /// connection attempt.
+    pub dial_concurrency_factor: u8,
+
+    /// Local address.
+    pub listening_multiaddr: Multiaddr,
+
+    /// Maximum incoming connections.
+    pub max_connections_in: u32,
+
+    /// Maximum outgoing connections.
+    pub max_connections_out: u32,
+
+    /// Maximum pending incoming connections.
+    pub max_connections_pending_in: u32,
+
+    /// Maximum pending outgoing connections.
+    pub max_connections_pending_out: u32,
+
+    /// Maximum connections per peer (includes outgoing and incoming).
+    pub max_connections_per_peer: u32,
+
+    /// mDNS discovery enabled.
+    pub mdns: bool,
+
+    /// Notify handler buffer size.
+    ///
+    /// Defines the buffer size for events sent from the NetworkBehaviour to the ConnectionHandler.
+    /// If the buffer is exceeded, the Swarm will have to wait. An individual buffer with this
+    /// number of events exists for each individual connection.
+    pub notify_handler_buffer_size: usize,
+
+    /// Connection event buffer size.
+    ///
+    /// Defines the buffer size for events sent from the ConnectionHandler to the
+    /// NetworkBehaviour. Each connection has its own buffer. If the buffer is
+    /// exceeded, the ConnectionHandler will sleep.
+    pub per_connection_event_buffer_size: usize,
+
+    /// Ping behaviour enabled.
+    pub ping: bool,
+}
+
+impl Default for Libp2pConfiguration {
+    fn default() -> Self {
+        // Define the default listening multiaddress using the default QUIC port
+        let listening_multiaddr = format!("/ip4/0.0.0.0/udp/{QUIC_PORT}/quic-v1")
+            .parse()
+            .unwrap();
+
+        Self {
+            dial_concurrency_factor: 8,
+            listening_multiaddr,
+            max_connections_in: 16,
+            max_connections_out: 16,
+            max_connections_pending_in: 8,
+            max_connections_pending_out: 8,
+            max_connections_per_peer: 8,
+            mdns: false,
```

Comment:
I like that all behaviour is turned off by default, for the `aquadoggo_cli` I can imagine we want it to be turned on by overwriting that default.